### PR TITLE
Add refactored Music animator code into master

### PR DESCRIPTION
**Pull request by jamieyu96**

### CODE REVIEW

File: music-animator/spotify.js [hunk 1]
```diff
@@ -0,0 +1,216 @@
+const queryInput = document.querySelector('#query')
```
Comment (jamieyu96) [inline music-animator/spotify.js]: Let's keep anything interacting with HTML in script.js.

queryInput as string parameter into startMusic()

File: music-animator/spotify.js [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+  return fetch('https://spotify-web-api-token.herokuapp.com/token')
+    .then(function(response) {
+      return response.json();
+    })
+    .then(function(myJson) {
+      return myJson.token;
+    })
+    .catch(function(err) {
+      console.log(err);
+      debugger;
```
Comment (jamieyu96) [inline music-animator/spotify.js]: Remove

File: music-animator/spotify.js [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+          offlineContext.startRendering();
+        });
+
+        offlineContext.oncomplete = function(audio) {
+          animator(audio);
+        };
+      };
+
+      request.send();
+    });
```
Comment (jamieyu96) [inline music-animator/spotify.js]: Pull this out into a separate function here. Maybe called requestMusic()?

File: music-animator/spotify.js [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+function startMusic(animator) {
+	queryInput.remove();
+
+  if (!audioTag.paused) {
+    audioTag.pause();
+		document.getElementById('playCirclePlaying').setAttribute("id", "playCircle");
+  } else if (audioTag.paused) {
+    audioTag.play();
+		document.getElementById('playCircle').setAttribute("id", "playCirclePlaying");
+  }
```
Comment (jamieyu96) [inline music-animator/spotify.js]: Pull this back into script.js, this should be a function called playMusic() which we can have them write. Then playMusic() will call the requestMusic() function from this class.

File: music-animator/spotify.js (hunk 3 above)
Comment (adc613) [reply music-animator/spotify.js]: Not sure how to pull this out

**Comment by adc613** [comment]
Reminder to myself that there's an issue that creating an infinite loop when I play a song

**Pull request merged**